src/io/flutter/run/LauncherState.java

```diff
@@ -238,33 +254,71 @@ protected boolean canRun(C config) {
     @Override
     protected final RunContentDescriptor doExecute(@NotNull RunProfileState state, @NotNull ExecutionEnvironment env)
       throws ExecutionException {
-      if (!(state instanceof Launcher)) {
+      if (!(state instanceof LauncherState)) {
         LOG.error("unexpected RunProfileState: " + state.getClass());
         return null;
       }
-      final Launcher launcher = (Launcher)state;
-      return launcher.launch(env);
+
+      final LauncherState launcherState = (LauncherState)state;
+      final String executorId = env.getExecutor().getId();
+
+      // See if we should issue a hot-reload.
+      final List<RunContentDescriptor> runningProcesses =
+        ExecutionManager.getInstance(env.getProject()).getContentManager().getAllDescriptors();
+
+      for (RunContentDescriptor descriptor : runningProcesses) {
+        final ProcessHandler process = descriptor.getProcessHandler();
+        if (process != null && !process.isProcessTerminated() && process.getUserData(FLUTTER_RUN_CONFIG_KEY) == launcherState.runConfig) {
+          final String launchMode = process.getUserData(FLUTTER_LAUNCH_MODE_KEY);
```

Comment:
Seems like this would be simpler if we had a getRunningApp() method - see below.